src/main/resources/assets/carpet/scripts/shapes.scl

```diff
@@ -1,15 +1,20 @@
 //This is a library app, mostly for the draw command, which allows to import these shape-drawing functions into your own app.
-//Arguments for each are to be given as a singleton list, for ease of use. This is with the exception of the fill_flat functions, with
-//the assumption that they're unlikely to be used outside of internal calls, but it is still available to be called if necessary.
-//Make sure you read the arguments and input them in the correct order, or it will mess up.
-//these functions return a set of all the positions in the shape, given the parameters, without repeating a position.
+//The _fill_flat functions are mostly used in internal calls, but can be used outside as well
+//These functions return a set of all the positions in the shape, given the parameters, without repeating a position.
 //This way, you can efficiently iterate over the positions once having generated them. See draw_beta.sc for an example. todo change to draw once renamed
+//You can also read the documentation code above each function to see what it does, how it does it and how well it does it
+//One way you can speed things up is by threading the operations (see draw_beta.sc)
 
-
-draw_sphere(args)->(
-    [centre, radius, hollow] = args;
+//This algorithm turns half a circle (in radians), then at each angle
+//it turns another half circle in radians and then fills in the other half using - signs
+//For a filled circle we just fill in the end points of the circle
+//Algorithmic efficiency:
+//  O(n^2) for hollow spheres cos 2d surface
+//  O(n^3) for filled spheres cos 3d volume
+draw_sphere(centre, radius, hollow)->(
```

Comment:
But you don't say what it does! Explaining how it works is nice, but it's missing what it does.

Reply:
err... got caught up in the explanation, but will do